File: sdk/typescript/src/control_plane.ts
/**
 * TypeScript SDK Client for Enterprise Control Plane & Platform Management.
 */

export class ControlPlaneClient {
  private baseUrl: string;

  constructor(baseUrl: string = "http://localhost:8002") {
    this.baseUrl = baseUrl.replace(/\/$/, "");
  }

  async getSummary(): Promise<Record<string, any>> {
    const res = await fetch(`${this.baseUrl}/v1/control-plane/summary`);
    if (!res.ok) throw new Error(`Summary fetch failed: ${res.statusText}`);
    return res.json();
  }

  async listTenants(): Promise<Record<string, any>> {
    const res = await fetch(`${this.baseUrl}/v1/control-plane/tenants`);
    if (!res.ok) throw new Error(`Tenants fetch failed: ${res.statusText}`);
    return res.json();
  }
}

File: sdk/go/control_plane.go
package llmengine

import (
	"encoding/json"
	"fmt"
	"net/http"
)

type ControlPlaneClient struct {
	BaseURL string
	HTTPClient *http.Client
}

func NewControlPlaneClient(baseURL string) *ControlPlaneClient {
	return &ControlPlaneClient{
		BaseURL: baseURL,
		HTTPClient: &http.Client{},
	}
}

func (c *ControlPlaneClient) GetSummary() (map[string]interface{}, error) {
	resp, err := c.HTTPClient.Get(fmt.Sprintf("%s/v1/control-plane/summary", c.BaseURL))
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var result map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
		return nil, err
	}
	return result, nil
}
